### ISSUE TITLE
Steam does not launch; errors out with "Assertion Failed"

### ISSUE THREAD

**Issue by dylanJasperX1** [Botspot/pi-apps#2517]
### Confirmations

- [X] I searched for similar bug reports in https://github.com/Botspot/pi-apps/issues and found none were relevant.
- [X] I tried updating Pi-Apps and the issue is still not fixed. [How to update Pi-Apps?](https://pi-apps.io/wiki/getting-started/updating-pi-apps/)
- [X] IMPORTANT: I promise to respond to comments on this issue and write a good bug report https://www.softwaretestinghelp.com/how-to-write-good-bug-report/

### What happened?

Steam seems to install without issue, but fails to launch. Opening from the terminal results in "Assertion Failed" error

### Description

I have been attempting to install steam from pi apps, but have been unable to get it to launch. I am using a Raspberry Pi 5 running Debian 12, although I also experinced this issue when I tried this on Ubuntu 23.10. When I try to run steam from the terminal, I get an error. The output ends with errors "Assertion Failed" and messages about stopping "waiting for remaining thread". Before installing steam, the only programs I had installed were pi apps and OBS to record the video linked below. I was also using a 4k PageSize Linux Kernel.

A video of the install process is provided [here](https://drive.google.com/file/d/1EFYfvckqLyTLom3dk8_PTa0pjrN_dHLg/view?usp=sharing).

**Steps to reproduce**
1. Boot a fresh install of Debian 12.
2. Install pi apps
3. Install steam from pi apps. When asked to switch to a 4k pagesize kernel, select "Yes, switch to 4K PageSize Kernel" 
4. After the system restarts, reopen pi apps and install steam again.
5. After steam finishes downloading and installing updates, launch steam from the terminal.

**Expected result**
Steam launches

**Actual result**
The error log provided with this bug report.

I apologize if this is not a good bug report. I've never actually done something like this before.
  

### What are your system specs (run the following command in your terminal)?

```bash
OS: Debian GNU/Linux 12 (bookworm)
OS architecture: 64-bit
Last updated Pi-Apps on: 01/06/2024
Latest Pi-Apps version: 01/06/2024
Kernel: aarch64 6.1.0-rpi7-rpi-v8
Device model: Raspberry Pi 5 Model B Rev 1.0
SOC identifier: bcm2712
Cpu name: Cortex-A76
Ram size: 7.94 GB
Raspberry Pi OS image version: 2023-12-05
Language: en_US.UTF-8
```

### (Recommended) Error log? Terminal output? Debug messages?

```bash
steam.sh[37946]: Running Steam on debian 12 64-bit
steam.sh[37946]: STEAM_RUNTIME is enabled by the user
setup.sh[38011]: Steam runtime environment up-to-date!
steam.sh[37946]: Steam client's requirements are satisfied
Debug level is 1
Dynarec for ARMv8, with extension: HALF FAST_MULT EDSP NEON VFPv4 IDIVA AES PMULL PageSize:4096 Running on Cortex-A76 with 4 Cores
Params database has 29 entries
Box86 with Dynarec v0.3.5 fbf3e1b1 built on Jan  5 2024 06:26:39
Using default BOX86_LD_LIBRARY_PATH: ./:lib/:lib32/:x86/:i686/
Using default BOX86_PATH: ./:bin/
Counted 65 Env var
Looking for /home/dylan/.local/share/Steam/ubuntu12_32/steam
steam detected
Apply RC params for steam
Applying BOX86_NOGTK=1
Applying BOX86_DYNAREC_STRONGMEM=1
Applying BOX86_DYNAREC_CALLRET=1
Applying BOX86_EMULATED_LIBS=libudev.so.0:libSDL2-2.0.so.0
argv[1]="-no-cef-sandbox"
argv[2]="steam://open/minigameslist"
Rename process to "steam"
Using native(wrapped) libdl.so.2
Using native(wrapped) libc.so.6
Using native(wrapped) ld-linux.so.2
Using native(wrapped) libpthread.so.0
Using native(wrapped) librt.so.1
Using native(wrapped) libbsd.so.0
Using native(wrapped) libX11.so.6
Using native(wrapped) libm.so.6
[2024-01-06 22:15:13] Startup - updater built Dec  8 2023 00:32:59
[2024-01-06 22:15:13] Startup - Steam Client launched with: '/home/dylan/.local/share/Steam/ubuntu12_32/steam' '-no-cef-sandbox' 'steam://open/minigameslist'
Using native(wrapped) crashhandler.so
Using native(wrapped) libGL.so.1
Looks like steam didn't shutdown cleanly, scheduling immediate update check
[2024-01-06 22:15:14] Loading cached metrics from disk (/home/dylan/.local/share/Steam/package/steam_client_metrics.bin)
[2024-01-06 22:15:14] Failed to load cached hosts file (File 'update_hosts_cached.vdf' not found), using defaults
[2024-01-06 22:15:14] Using the following download hosts for Public, Realm steamglobal
[2024-01-06 22:15:14] 1. https://cdn.steamstatic.com, /client/, Realm 'steamglobal', weight was 1, source = 'baked in'
[2024-01-06 22:15:14] Checking for update on startup
[2024-01-06 22:15:14] Checking for available updates...
[2024-01-06 22:15:14] Downloading manifest: https://cdn.steamstatic.com/client/steam_client_ubuntu12
[2024-01-06 22:15:14] Manifest download: send request
[2024-01-06 22:15:14] Manifest download: waiting for download to finish
[2024-01-06 22:15:15] Manifest download: finished
[2024-01-06 22:15:15] Download skipped: /client/steam_client_ubuntu12 version 1702079146, installed version 1702079146, existing pending version 0
[2024-01-06 22:15:15] Nothing to do
[2024-01-06 22:15:15] Verifying installation...
[2024-01-06 22:15:15] Performing checksum verification of executable files
[2024-01-06 22:15:16] Verification complete
Using emulated /home/dylan/.local/share/Steam/ubuntu12_32/steam-runtime/usr/lib/i386-linux-gnu/libappindicator.so.1
Using emulated /home/dylan/.local/share/Steam/ubuntu12_32/steam-runtime/usr/lib/i386-linux-gnu/libindicator.so.7
Using emulated /home/dylan/.local/share/Steam/ubuntu12_32/steam-runtime/pinned_libs_32/libdbusmenu-gtk.so.4
Using emulated /home/dylan/.local/share/Steam/ubuntu12_32/steam-runtime/pinned_libs_32/libdbusmenu-glib.so.4
Using emulated /home/dylan/.local/share/Steam/ubuntu12_32/steam-runtime/pinned_libs_32/libgtk-x11-2.0.so.0
Using emulated /home/dylan/.local/share/Steam/ubuntu12_32/steam-runtime/usr/lib/i386-linux-gnu/libgio-2.0.so.0
Using emulated /home/dylan/.local/share/Steam/ubuntu12_32/steam-runtime/usr/lib/i386-linux-gnu/libgobject-2.0.so.0
Using emulated /home/dylan/.local/share/Steam/ubuntu12_32/steam-runtime/lib/i386-linux-gnu/libglib-2.0.so.0
Using emulated /home/dylan/.local/share/Steam/ubuntu12_32/steam-runtime/usr/lib/i386-linux-gnu/libgdk_pixbuf-2.0.so.0
Using emulated /home/dylan/.local/share/Steam/ubuntu12_32/steam-runtime/usr/lib/i386-linux-gnu/libgmodule-2.0.so.0
Using emulated /home/dylan/.local/share/Steam/ubuntu12_32/steam-runtime/usr/lib/i386-linux-gnu/libgdk-x11-2.0.so.0
Using emulated /home/dylan/.local/share/Steam/ubuntu12_32/steam-runtime/usr/lib/i386-linux-gnu/libpangocairo-1.0.so.0
Using native(wrapped) libXfixes.so.3
Using native(wrapped) libxcb.so.1
Using native(wrapped) libXau.so.6
Using native(wrapped) libXdmcp.so.6
Using emulated /home/dylan/.local/share/Steam/ubuntu12_32/steam-runtime/usr/lib/i386-linux-gnu/libatk-1.0.so.0
Using native(wrapped) libcairo.so.2
Using emulated /home/dylan/.local/share/Steam/ubuntu12_32/steam-runtime/usr/lib/i386-linux-gnu/libpangoft2-1.0.so.0
Using emulated /home/dylan/.local/share/Steam/ubuntu12_32/steam-runtime/usr/lib/i386-linux-gnu/libpango-1.0.so.0
Using native(wrapped) libfontconfig.so.1
Using native(wrapped) libexpat.so.1
Using native(wrapped) libfreetype.so.6
Using native(wrapped) libXext.so.6
Using native(wrapped) libXrender.so.1
Using native(wrapped) libXinerama.so.1
Using native(wrapped) libXi.so.6
Using native(wrapped) libXrandr.so.2
Using native(wrapped) libXcursor.so.1
Using native(wrapped) libXcomposite.so.1
Using native(wrapped) libXdamage.so.1
Using emulated /home/dylan/.local/share/Steam/ubuntu12_32/steam-runtime/usr/lib/i386-linux-gnu/libffi.so.6
Error initializing native libpcre.so.3 (last dlerror is libpcre.so.3: wrong ELF class: ELFCLASS64)
Using emulated /home/dylan/.local/share/Steam/ubuntu12_32/steam-runtime/lib/i386-linux-gnu/libpcre.so.3
Error initializing native libpng12.so.0 (last dlerror is libpng12.so.0: wrong ELF class: ELFCLASS64)
Using emulated /home/dylan/.local/share/Steam/ubuntu12_32/steam-runtime/lib/i386-linux-gnu/libpng12.so.0
Using emulated /home/dylan/.local/share/Steam/ubuntu12_32/steam-runtime/usr/lib/i386-linux-gnu/libtiff.so.4
Using native(wrapped) libjpeg.so.8
Using native(wrapped) libz.so.1
Using native(wrapped) libselinux.so.1
Using native(wrapped) libresolv.so.2
Look for __dn_expand in loaded elfs
Found symbol with version GLIBC_2.4, value = 0xf7e766a4
Look for __res_query in loaded elfs
Found symbol with version GLIBC_2.4, value = 0xf7e7e534
Using emulated /home/dylan/.local/share/Steam/ubuntu12_32/steamui.so
Using emulated /home/dylan/.local/share/Steam/ubuntu12_32/libtier0_s.so
Using emulated /home/dylan/.local/share/Steam/ubuntu12_32/libvstdlib_s.so
Using emulated /home/dylan/.local/share/Steam/ubuntu12_32/libSDL3.so.0
Mamp64 for (@0xe8976000 0x8ea598) for elf "/home/dylan/.local/share/Steam/ubuntu12_32/libv8.so" returned 0xe8976000 insteadUsing emulated /home/dylan/.local/share/Steam/ubuntu12_32/libv8.so
Mamp64 for (@0xe83b2000 0x4c5520) for elf "/home/dylan/.local/share/Steam/ubuntu12_32/libavcodec.so.58" returned 0xe83b2000 insteadUsing emulated /home/dylan/.local/share/Steam/ubuntu12_32/libavcodec.so.58
Mamp64 for (@0xe80da000 0x2d7120) for elf "/home/dylan/.local/share/Steam/ubuntu12_32/libavutil.so.56" returned 0xe80da000 insteadUsing emulated /home/dylan/.local/share/Steam/ubuntu12_32/libavutil.so.56
Mamp64 for (@0xe7a6f000 0x66acd8) for elf "/home/dylan/.local/share/Steam/ubuntu12_32/libvideo.so" returned 0xe7a6f000 insteadUsing emulated /home/dylan/.local/share/Steam/ubuntu12_32/libvideo.so
Using native(wrapped) libXtst.so.6
Using emulated /home/dylan/.local/share/Steam/ubuntu12_32/steam-runtime/usr/lib/i386-linux-gnu/libpipewire-0.3.so.0
Using native(wrapped) libpulse.so.0
Using native(wrapped) libX11-xcb.so.1
Using emulated /home/dylan/.local/share/Steam/ubuntu12_32/steam-runtime/usr/lib/i386-linux-gnu/libxcb-res.so.0
Warning: Weak Symbol _ZGTtnaj not found, cannot apply R_386_JMP_SLOT 0x646b8dd0 (0xd536)
Warning: Weak Symbol _ZGTtnaj not found, cannot apply R_386_JMP_SLOT 0x64766aa4 (0xb1c6)
Using emulated /home/dylan/.local/share/Steam/ubuntu12_32/libicui18n.so
Using emulated /home/dylan/.local/share/Steam/ubuntu12_32/libicuuc.so
Using emulated /home/dylan/.local/share/Steam/ubuntu12_32/steam-runtime/usr/lib/i386-linux-gnu/libstdc++.so.6
Using emulated /home/dylan/.local/share/Steam/ubuntu12_32/steam-runtime/lib/i386-linux-gnu/libgcc_s.so.1
Using emulated /home/dylan/.local/share/Steam/ubuntu12_32/libvpx.so.6
Using emulated /home/dylan/.local/share/Steam/ubuntu12_32/steam-runtime/usr/lib/i386-linux-gnu/libva.so.2
Using emulated /home/dylan/.local/share/Steam/ubuntu12_32/steam-runtime/usr/lib/i386-linux-gnu/libvdpau.so.1
Using native(wrapped) libdrm.so.2
Using emulated /home/dylan/.local/share/Steam/ubuntu12_32/libavformat.so.58
Using emulated /home/dylan/.local/share/Steam/ubuntu12_32/libavresample.so.4
Using emulated /home/dylan/.local/share/Steam/ubuntu12_32/libswscale.so.5
Error initializing native libbz2.so.1 (last dlerror is libbz2.so.1: wrong ELF class: ELFCLASS64)
Using emulated /home/dylan/.local/share/Steam/ubuntu12_32/steam-runtime/lib/i386-linux-gnu/libbz2.so.1.0
Warning: Weak Symbol _ZGTtnaj not found, cannot apply R_386_JMP_SLOT 0xe809247c (0x35fb6)
Warning: Weak Symbol _ZGTtnaj not found, cannot apply R_386_JMP_SLOT 0x64618b74 (0x114cf6)

Steam logging initialized: directory: /home/dylan/.local/share/Steam/logs

38038|SIGSEGV @0xe36eebac (???(0xe36eebac)) (x86pc=0xe36e9148/???:"???", esp=0xf731bd40, stack=0xf6b23000:0xf7323000 own=(nil) fp=0xc), for accessing (nil) (code=1/prot=0), db=0xe36fdf30(0xe36eebac:0xe36eed10/0xe36e9148:0xe36e9164/???:clean, hash:80af3df6/80af3df6)
EAX:0x00000000 ECX:0x65ae15fc EDX:0xe6b39a00 EBX:0x6114dff4 
ESP:0xf731bd40 EBP:0xe44bf7a0 ESI:0x658df950 EDI:0xe6b39a00 
ESP-0x10:0x658df950 ESP-0x0c:0xe6b39a00 ESP-0x08:0xe44bf7a0 ESP-0x04:0x6111b3be
ESP+0x00:0x00000000 ESP+0x04:0x613f7ff4 ESP+0x08:0x00000004 ESP+0x0c:0x6114dff4
38038|Double SIGSEGV (code=1, pc=0xe36eebac, addr=(nil))!
src/common/framefunction.cpp (238) : Assertion Failed: CFrameFunctionMgr::~CFrameFunctionMgr: non static FrameFunction[CHTTPClient::BFrameFuncHandleCompletedWorkItems] still registered
src/common/framefunction.cpp (238) : Assertion Failed: CFrameFunctionMgr::~CFrameFunctionMgr: non static FrameFunction[CHTTPClient::BFrameFuncHandleCompletedWorkItems] still registered
Using native(wrapped) crashhandler.so
Stop waiting for remaining thread 38045
Stop waiting for remaining thread 38046
```

**Comment by theofficialgman** [comment]
There is already another ticked open for steam launch failures https://github.com/Botspot/pi-apps/issues/2486 as well as an upstream bug report https://github.com/ptitSeb/box64/issues/1177 at box64

the error you get is different though and indicates a bug in box86 (error in running x86 code and not x86_64 code). Please open an issue upstream at the box86 repo https://github.com/ptitSeb/box86/issues

**Comment by DatBoiG265** [comment]
I have the same problem as well when trying to install steam on Raspberry Pi OS Bookworm 64bit

**Comment by vasi** [comment]
Does this fix help you? https://github.com/ptitSeb/box86/issues/920#issuecomment-1895845434

**Comment by theofficialgman** [comment]
> Does this fix help you? [ptitSeb/box86#920 (comment)](https://github.com/ptitSeb/box86/issues/920#issuecomment-1895845434)

@vasi don't create a custom .box86rc. latest box86 has the fix, simply apt update and upgrade.

**Comment by theofficialgman** [comment]
this issue was fixed with box86 last week https://github.com/ptitSeb/box86/commit/2b35400dc5d828bbb348f4527aa790392b51f7ba
